### ISSUE TITLE
Detecting empty sets

### ISSUE THREAD

**Issue by daemonfire300** [charlierguo/gmail#16]
I decided to write a little email filter to run on my vps and I ran into an interesting problem:

> Traceback (most recent call last):
> File "mailfilter.py", line 19, in <module>
>  mail.read()
> File "/usr/local/lib/python2.7/dist-packages/gmail-0.0.5-py2.7.egg/gmail/message.py", line 41, in read
>   self.gmail.imap.uid('STORE', self.uid, '+FLAGS', flag)
> File "/usr/lib/python2.7/imaplib.py", line 760, in uid
>   typ, dat = self._simple_command(name, command, *args)
> File "/usr/lib/python2.7/imaplib.py", line 1070, in _simple_command
>   return self._command_complete(name, self._command(name, *args))
> File "/usr/lib/python2.7/imaplib.py", line 905, in _command_complete
>   raise self.error('%s command error: %s %s' % (name, typ, data))
> imaplib.error: UID command error: BAD ['Could not parse command']

https://gist.github.com/anonymous/bd96cbcd768ce9c2647d
This is my code.

The problem seems to be that after running the filter once, it works as expected. The unwanted behaviour pops up if you run it multiple times if there are no items left matching the criteria, instead of holding 0 or None emails contains 1 message object, which I checked many times against my actual gmail account is not there. This falls automatically through the len() check and causes problems if you start calling methods on this "fake" message object. With message_id = None and no uid.

**Comment by pgwhalen** [comment]
So to be clear, if you run all of that code once it works (including the for loop), but if you run some portion of it again, that's when it breaks?  If understand correctly, there's a small chance it could be related to issue #17.  I found that to be the an issue myself when running the same code twice in a row.

Or otherwise, do you mean that `emails` always has that extra dummy message, even when running that code once?

**Comment by daemonfire300** [comment]
The process is the following:
- Check for potential emails by myself
- run the script on it (if it has emails matching the filter) it works fine
- if there are no emails matching the filter, the code breaks because emails only contains 1 element of type `message` with no attributes assigned. (everything is set to `None`) This break happens only if there are no emails matching the filter. I found this out by looking into my mail account while running the script.

This occured to me, when I ran my script multiple times today.
